test/test_overrides.py

```diff
@@ -488,5 +488,45 @@ def test(self):
 
 generate_tensor_like_override_tests(TestTorchFunctionOverride)
 
+class TestEinsumOverride(TestCase):
+    "Regression test for gh-38479"
+    def test_wrapper(self):
+        class Wrapper():
+            "Basic data container that knows how to unwrap itself"
+            def __init__(self, data):
+                self.data = data
+
+            def __torch_function__(self, func, types, args=(), kwargs=None):
+                if kwargs is None:
+                    kwargs = {}
+
+                # unwrap inputs if necessary
+                def unwrap(v):
+                    return v.data if isinstance(v, Wrapper) else v
+
+                args = map(unwrap, args)
+                kwargs = {k: unwrap(v) for k, v in kwargs.items()}
+
+                return func(*args, **kwargs)
+
+        x = Wrapper(torch.randn(5))
+        y = Wrapper(torch.randn(4))
+        self.assertTrue(torch.allclose(torch.einsum('i,j->ij', x, y),
+                                       torch.ger(x, y)))
+
+        # in the old einsum interface, `operands` is a list
+        a = Wrapper(torch.randn(2, 3))
+        b = Wrapper(torch.randn(5, 3, 7))
+        c = Wrapper(torch.randn(2, 7))
+        self.assertTrue(torch.allclose(torch.einsum('ik,jkl,il->ij', [a, b, c]),
+                                       torch.nn.functional.bilinear(a, c, b)))
+
+        # `equation` input is wrapped
+        As = Wrapper(torch.randn(3, 2, 5))
+        Bs = Wrapper(torch.randn(3, 5, 4))
+        equation = Wrapper('bij,bjk->bik')
```

Comment:
I don’t think non-tensor-likes should be wrapped. This gives the impression that they will be dispatched on, which they won’t.

Reply:
Fair enough, I'll remove that test case.